### DISCUSSION TITLE
Failed to load resource: the server responded with a status of 404 (Not Found)

### DISCUSSION THREAD

**Discussion by ghost** [iTwin/itwinjs-core#5815] [General]
The iTwin Viewer application compiled using npm run build。
When starting the iTwin Viewer program, it displays a failure to load resources. 
Are the resources in the locales directory generated automatically during the build, or do they need to be copied from somewhere else?
![resource_error](https://github.com/iTwin/itwinjs-core/assets/34014227/403569ba-33a3-4b5e-acfc-a4579a4b2f0e)
![build](https://github.com/iTwin/itwinjs-core/assets/34014227/bc8307bc-cf26-4e7e-8f4f-7935fcf3b996)

**Comment by aruniverse** [comment]
So iTwin.js only delivers localization files for `en`, [see](https://github.com/iTwin/itwinjs-core/blob/master/core/frontend/src/public/locales/en/CoreTools.json) as an example 

By default, in the Viewer, we just use the default [ITwinLocalization client](https://github.com/iTwin/itwinjs-core/blob/master/core/i18n/src/ITwinLocalization.ts#L33C14-L33C31) which checks what locales to use based on [detection.order](https://github.com/iTwin/itwinjs-core/blob/master/core/i18n/src/ITwinLocalization.ts#L50). Based on this config, it sees that you're using `zh-CN`, for which we don't deliver our own localization files for

**Reply by ghost** [reply]
My frontend program used viewer component from "@itwin/desktop-viewer-react"。
（1）Does the viewer component do IModelApp.localization automatically? do I need to go IModelApp.localization in my startup?
（2）When I build the program using react-scripts, the "locales" directory is not generated. Why is this happening? 

Thank you very much for your help. @aruniverse 
![nolocales](https://github.com/iTwin/itwinjs-core/assets/34014227/7e8e5e6e-20bc-488b-b9c0-e8d15b9e0790)
